src/main/scala/dspjunctions/JtagAxi.scala

```diff
@@ -0,0 +1,126 @@
+// See LICENSE for license details.
+
+package dspjunctions
+
+import cde._
+import chisel3._
+import chisel3.util._
+import jtag._
+import junctions._
+
+case object DspChainIncludeJtag extends Field[Bool]
+
+class DspJtagIO extends JtagIO {
+  // val TRST = Input(Bool())
+}
+
+trait HasDspJtagParameters {
+  implicit val p: Parameters
+  val includeJtag = try {
+    if (p(DspChainIncludeJtag)) {
+      true
+    } else {
+      false
+    }
+  } catch {
+    case _: ParameterUndefinedException => false
+  }
+}
+
+trait HasDspJtagIO extends HasDspJtagParameters {
+  val p: Parameters
+  val jtag = if (includeJtag) {
+      Some(new DspJtagIO)
+  } else {
+    None
+  }
+}
+
+trait HasDspJtagModule extends HasDspJtagParameters {
+  val io: HasDspJtagIO
+
+  val jtagMaster = if (includeJtag) {
+    Some(Module(new JtagAxiMaster()))
+  } else {
+    None
+  }
+
```

Comment:
Seems kinda janky. Do we need a parameter AND a mixin? Can the mixin add the parameter or something?